src/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/SyndicationElementExtension.cs

```diff
@@ -271,13 +272,8 @@ public void WriteTo(XmlWriter writer)
 
             internal void ComputeOuterNameAndNs(out string name, out string ns)
             {
-                if (_outerName != null)
-                {
```

Comment:
There are two different _outerName variables. SyndicationElementExtension._outerName has already been checked for null, but the instance being checked here is ExtensionDataWriter._outerName which is a field in a different class. What this code does is if ExtensionDataWriter._outerName has already been set, it copies ExtensionDataWriter._outerName and ExtensionDataWriter._outerNamespace to the passed in out parameters which reference the matching field names in SyndicationElementExtension. This change will cause a perf regression as it caused the outer name and ns to be recalculated on every call instead of calculating it once and caching it.

Reply:
`_outerName` is a readonly property so is never set outside of the constructor
```cs
            public ExtensionDataWriter(object extensionData, XmlObjectSerializer dataContractSerializer, string outerName, string outerNamespace)
            {
                Debug.Assert(extensionData != null && dataContractSerializer != null, "null check");
                _dataContractSerializer = dataContractSerializer;
                _extensionData = extensionData;
                _outerName = outerName;
                _outerNamespace = outerNamespace;
            }
```

```cs
private readonly string _outerName;
```

This constructor is only called from the `SyndicationElementExtension` constructor:
```cs
        public SyndicationElementExtension(string outerName, string outerNamespace, object dataContractExtension, XmlObjectSerializer dataContractSerializer)
        {
            if (dataContractExtension == null)
            {
                throw new ArgumentNullException(nameof(dataContractExtension));
            }
            if (outerName == string.Empty)
            {
                throw new ArgumentException(SR.OuterNameOfElementExtensionEmpty, nameof(outerName));
            }

            if (dataContractSerializer == null)
            {
                dataContractSerializer = new DataContractSerializer(dataContractExtension.GetType());
            }
            _outerName = outerName;
            _outerNamespace = outerNamespace;
            _extensionData = dataContractExtension;
            _extensionDataWriter = new ExtensionDataWriter(_extensionData, dataContractSerializer, _outerName, _outerNamespace);
        }
```

So `ExtensionDataWriter._outerName` shadows `SyndicationElementExtension._outerName`

So the null check for `SyndicationElementExtension._outerName` is the same as the null check for `ExtensionDataWriter._outerName`

Reply:
You're right.